File: go-solutions/1679_max_operations.go
package leetcode

import (
	"sort"
)

func MaxOperations(nums []int, k int) int {
	sort.Ints(nums)
	result := 0
	left, right := 0, len(nums)-1
	for left < right {
		sum := nums[left] + nums[right]
		if sum == k {
			result++
			left++
			right--
		} else if sum > k {
			right--
		} else {
			left++
		}
	}
	return result
}

File: ts-solutions/src/1679-max-operations.ts
export const maxOperations = (nums: number[], k: number): number => {
	nums.sort((a, b) => a - b)
	let left = 0,
		right = nums.length - 1,
		count = 0
	while (left < right) {
		const sum = nums[left] + nums[right]
		if (sum < k) {
			left++
		} else if (sum > k) {
			right--
		} else {
			left++
			right--
			count++
		}
	}
	return count
}
